frontend/ui/widget/widget.lua

```diff
@@ -14,44 +14,42 @@ local EventListener = require("ui/widget/eventlistener")
 
 --- Widget base class
 -- @table Widget
-local Widget = EventListener:new()
+local Widget = EventListener:extend{}
 
 --[[--
-Use this method to define a subclass widget class that's inherited from a
-base class widget. It only setups the metabale (or prototype chain) and will
-not initiate a real instance, i.e. call self:init().
+Use this method to define a widget subclass that's inherited from a base class widget.
+It only setups the metatable (or prototype chain) and will not initiate a real instance, i.e. call self:init().
 
-@tparam Widget baseclass
+@tparam table subclass
 @treturn Widget
 ]]
-function Widget:extend(baseclass)
-    local o = baseclass or {}
+function Widget:extend(subclass)
+    local o = subclass or {}
```

Comment:
(The name `baseclass` was wrong (didn't ever noticed).
But `subclass` although more correct, feels yet a bit wrong : we pass it a table/prototype (with default properties) and it is returned as being upgraded to be a subclass.
But ok, we pass what we want to become a subclass.)

Reply:
Yeah, I didn't think of anything better ;o).

(I'm not necessarily super fond of "extend" either, FWIW).

Reply:
But I kinda like `prototype`, so... maybe `subclass_prototype` would work better?